### PR TITLE
feat(Manager) : environment filter in spa details page

### PR DESCRIPTION
**Pull request by pjain24**
## Closes / Fixes / Resolves
Closes #issue_id

## Explain the feature/fix
Implemented an "Environment Filter" on the SPA Detailed Page, which allows stakeholders to filter and view all SPAs within a specific environment. This feature provides a convenient way for stakeholders to search and access relevant information, enhancing the user experience.

## Does this PR introduce a breaking change
No

## Screenshot(s)

<details>
<summary>View Screenshots</summary>

277949327-206e7953-96b3-410e-8a16-9aa24f5fc36e
277949169-25bcc541-08c3-4755-a1d9-58af17a38437
277949034-1ba1add5-e3fc-4722-bfc7-d06c7b62dd47

</details>

### CODE REVIEW

File: packages/manager/src/views/WebPropertyDetailPage/components/SSR/StaticDeployment.tsx [hunk 1]
```diff
@@ -86,213 +97,294 @@ export const StaticDeployment = () => {
...
-              <Th textCenter>Router URL</Th>
-              <Th textCenter style={{ justifyContent: 'space-evenly', display: 'grid' }}>
-                Actions
-              </Th>
-            </Tr>
-          </Thead>
+    <>
+      <Split hasGutter className="pf-u-mb-md">
+        <SplitItem isFilled>
+          <Select
```
Comment (nikhitamore) [inline packages/manager/src/views/WebPropertyDetailPage/components/SSR/StaticDeployment.tsx:126]: Can you write this in one component and use it in both containerized and static , as i can see code is being repeated in both files.

File: packages/manager/src/views/WebPropertyDetailPage/components/SSR/ContainerizedDeployment.tsx [hunk 2]
```diff
@@ -286,228 +296,258 @@ export const ContainerizedDeployment = () => {
...
-                <Td style={{ maxWidth: '15ch', wordWrap: 'break-word' }}>{`${val?.path.slice(
-                  0,
-                  URL_LENGTH_LIMIT
-                )} ${val?.path.length > URL_LENGTH_LIMIT ? '...' : ''}`}</Td>
-                <Td style={{ maxWidth: '15ch', wordWrap: 'break-word' }}>
-                  {`${val?.healthCheckPath.slice(0, URL_LENGTH_LIMIT)} ${
-                    val?.healthCheckPath.length > URL_LENGTH_LIMIT ? '...' : ''
-                  }`}
-                </Td>
+        <SplitItem isFilled>
```
Comment (nikhitamore) [inline packages/manager/src/views/WebPropertyDetailPage/components/SSR/ContainerizedDeployment.tsx:504]: Can you add a condition when there is no deployment then env filter should not be visible , so when empty state is there then env filter should not be visible

File: packages/manager/src/views/WebPropertyDetailPage/components/SSR/StaticDeployment.tsx (hunk 1 above)
Comment (pjain24) [reply packages/manager/src/views/WebPropertyDetailPage/components/SSR/StaticDeployment.tsx:126]: Ok

File: packages/manager/src/views/WebPropertyDetailPage/components/SSR/ContainerizedDeployment.tsx (hunk 2 above)
Comment (pjain24) [reply packages/manager/src/views/WebPropertyDetailPage/components/SSR/ContainerizedDeployment.tsx:504]: Okay

Comment (pjain24) [reply packages/manager/src/views/WebPropertyDetailPage/components/SSR/ContainerizedDeployment.tsx:504]: done

**Pull request closed without merging**